arcgis-ios-sdk-samples/Maps/Generate offline map/GenerateOfflineMapViewController.swift

```diff
@@ -248,15 +248,15 @@ class GenerateOfflineMapViewController: UIViewController, AGSAuthenticationManag
     private func showLoginQueryAlert() {
         
         let alertController = UIAlertController(title: nil, message: "This sample requires you to login in order to take the map's basemap offline. Would you like to continue?", preferredStyle: .alert)
-        let yesAction = UIAlertAction(title: "OK", style: .default) { [weak self] (action) in
+        let loginAction = UIAlertAction(title: "Login", style: .default) { [weak self] (action) in
```

Comment:
FYI: Since you're presenting the alert controller right away, you don't need a `weak` reference to `self` in the handler. `self` cannot go away while the alert controller is presented. And when it is dismissed, the alert controller is released, breaking any retain cycle.